pkg/services/irma/validator.go

```diff
@@ -22,40 +42,91 @@ import (
+const VerifiablePresentationType = "NutsIrmaPresentation"
+
+// ContractFormat holds the readable identifier of this signing means.
+const ContractFormat = "irma"
+
+// Service validates contracts using the irma logic.
+type Service struct {
+	IrmaSessionHandler SessionHandler
 	IrmaConfig         *irma.Configuration
-	Registry           registry.RegistryClient
-	Crypto             nutscrypto.Client
-	ContractTemplates  contract.TemplateStore
+	IrmaServiceConfig  ValidatorConfig
+	// todo: remove this when the deprecated ValidateJwt is removed
+	Registry          registry.RegistryClient
+	Crypto            nutscrypto.Client
+	ContractTemplates contract.TemplateStore
 }
 
-type IrmaServiceConfig struct {
-	Mode string
+// ValidatorConfig holds the configuration for the irma validator.
+type ValidatorConfig struct {
 	// Address to bind the http server to. Default localhost:1323
-	Address                   string
-	PublicUrl                 string
-	IrmaConfigPath            string
-	IrmaSchemeManager         string
+	Address string
+	// PublicURL is used for discovery for the IRMA app.
+	PublicURL string
+	// Where to find the IrmaConfig files including the schemas
+	IrmaConfigPath string
+	// Which scheme manager to use
+	IrmaSchemeManager string
+	// Auto update the schemas every x minutes or not?
 	SkipAutoUpdateIrmaSchemas bool
 }
 
-// LegacyIdentityToken is the JWT that was used as Identity token in versions prior to < 0.13
-type LegacyIdentityToken struct {
-	jwt.StandardClaims
-	Contract SignedIrmaContract `json:"nuts_signature"`
-}
-
 // IsInitialized is a helper function to determine if the validator has been initialized properly.
-func (v IrmaService) IsInitialized() bool {
+func (v Service) IsInitialized() bool {
 	return v.IrmaConfig != nil
 }
 
+// VerifiablePresentation is a specific proof for irma signatures
+type VerifiablePresentation struct {
+	contract.VerifiableCredentialBase
+	Proof VPProof `json:"proof"`
+}
+
+// VPProof is a specific IrmaProof for the specific VerifiablePresentation
+type VPProof struct {
+	contract.Proof
+	Signature string `json:"proofValue"`
+}
+
+// VerifyVP expects the given raw VerifiablePresentation to be of the correct type
+// todo: type check?
+func (v Service) VerifyVP(rawVerifiablePresentation []byte) (*contract.VerificationResult, error) {
+	// Extract the Irma message
+	vp := VerifiablePresentation{}
+	if err := json.Unmarshal(rawVerifiablePresentation, &vp); err != nil {
+		return nil, fmt.Errorf("could not verify VP: %w", err)
+	}
+
+	// Create the irma contract validator
+	contractValidator := contractVerifier{v.IrmaConfig, v.ContractTemplates}
+	signedContract, err := contractValidator.parseSignedIrmaContract(vp.Proof.Signature)
+	if err != nil {
+		return nil, err
+	}
+
+	cvr, err := contractValidator.verifyAll(signedContract, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// todo this is because of backwards compatibility
```

Comment:
what is todo? Code doesn't look like todo?

Reply:
dunno, @stevenvegt you?

Reply:
Neither do I, removed it. 🤞